### ISSUE TITLE
nnf_embedding - no argument input

### ISSUE THREAD

**Issue by jwijffels** [mlverse/torch#148]
``` r
library(torch)
x <- matrix(rnorm(20000*5), nrow = 20000, ncol = 5)
tt <- torch_tensor(x)

test <- nnf_embedding(torch_tensor(1:2, dtype = torch_long()), tt)
#> Error in torch_embedding(weight = weight, input = input, padding_idx = padding_idx, : unused argument (input = input)
test <- torch:::torch_embedding(weight = tt, 
                                indices = torch_tensor(0:1, dtype = torch_long()), 
                        padding_idx = -1L,
                        scale_grad_by_freq = FALSE, sparse = FALSE)
test <- as_array(test)
test
#>           [,1]       [,2]       [,3]       [,4]      [,5]
#> [1,] 0.1955841 -0.5570903 -0.2056061 -0.2091042 0.3444653
#> [2,] 1.0321622  1.5975115 -1.0296366 -0.3942826 1.0703566
x[1:2, ]
#>           [,1]       [,2]       [,3]       [,4]      [,5]
#> [1,] 0.1955841 -0.5570903 -0.2056061 -0.2091042 0.3444652
#> [2,] 1.0321622  1.5975116 -1.0296366 -0.3942826 1.0703566
all.equal(x[1:2, ], test)
#> [1] "Mean relative difference: 2.220236e-08"
```

<sup>Created on 2020-06-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by jwijffels** [comment]
On the other hand this 0-indexing does something different

``` r
x <- matrix(rnorm(2*5), nrow = 2, ncol = 5)
tt <- torch_tensor(x, dtype = torch_float())
#> Error in torch_tensor(x, dtype = torch_float()): could not find function "torch_tensor"
x
#>           [,1]       [,2]       [,3]       [,4]      [,5]
#> [1,] 0.6143988 -0.7305436 -0.3479258  1.3592944 0.7868064
#> [2,] 0.7401851 -0.9612402 -0.6420344 -0.1277194 0.2404325
tt[0, , drop = FALSE]
#> Error in eval(expr, envir, enclos): object 'tt' not found
tt[1, , drop = FALSE]
#> Error in eval(expr, envir, enclos): object 'tt' not found
tt[2, , drop = FALSE]
#> Error in eval(expr, envir, enclos): object 'tt' not found
```

<sup>Created on 2020-06-30 by the [reprex package](https://reprex.tidyverse.org) (v0.3.0)</sup>

**Comment by dfalbel** [comment]
Yeah, for now only the slice is 1-indexed, but we will move all indexing to 1-based in the future. 
Indexing with 0 when slicing should raise an error not return the first element . I openend #156 to track this.
We implemented `nn_embedding` in #155 and fixed the `input` argument in nnf_embedding.
